File: ExpertSystem.py
def classify_document(document):
    document = document.lower()  # Convert to lowercase for case-insensitive matching

    if "stock" in document or "finance" in document or "bank" in document:
        return "Finance"
    if "health" in document or "doctor" in document or "medicine" in document:
        return "Health"
    if "technology" in document or "computer" in document or "software" in document:
        return "Technology"
    return "Unknown Category"

# Main logic
document = input("Enter a brief document summary: ")
category = classify_document(document)
print("Predicted Category:", category)

File: ExpertSystem.cpp
#include <iostream>
#include <cstring>

using namespace std;

// Function to classify the document based on keywords
string classifyDocument(const char *document)
{
    if (strstr(document, "stock") || strstr(document, "finance") || strstr(document, "bank"))
    {
        return "Finance";
    }
    if (strstr(document, "health") || strstr(document, "doctor") || strstr(document, "medicine"))
    {
        return "Health";
    }
    if (strstr(document, "technology") || strstr(document, "computer") || strstr(document, "software"))
    {
        return "Technology";
    }
    return "Unknown Category";
}

int main()
{
    char document[200];

    cout << "Enter a brief document summary: ";
    cin.getline(document, 200);

    string category = classifyDocument(document);

    cout << "Predicted Category: " << category << endl;

    return 0;
}
